File: vb/simple.vb
Imports System.Net

Class Client
    Inherits WebClient
    Const username As String = "lum-customer-CUSTOMER-zone-YOURZONE"
    Const password As String = "YOURPASS"
    Const port = 22225
    Public session_id As String = New Random().Next().ToString()

    Public Sub New(Optional country As String = Nothing)
        Me.Proxy = New WebProxy("zproxy.luminati.io", port)
        Dim login = username &
            If(country IsNot Nothing, "-country-" & country, "") &
            "-session-" & session_id
        Me.Proxy.Credentials = New NetworkCredential(login, password)
    End Sub

    Protected Overrides Function GetWebRequest(address As Uri) As WebRequest
        Dim request = MyBase.GetWebRequest(address)
        request.ConnectionGroupName = session_id
        Return request
    End Function
End Class

Module Example
    Sub Main()
        Console.WriteLine("Performing request(s)")
        Dim session as New Client()
        ' Put full scraping sequence below:
        Console.WriteLine(session.DownloadString("http://lumtest.com/myip.json"))
        ' client.DownloadString(...second request...);
    End Sub
End Module

File: csharp/simple.cs

using System;
using System.Net;

class Client : WebClient
{
    public static string username = "lum-customer-CUSTOMER-zone-YOURZONE";
    public static string password = "YOURPASS";
    public static int port = 22225;
    public string session_id = new Random().Next().ToString();

    public Client(string country = null)
    {
        this.Proxy = new WebProxy("zproxy.luminati.io", port);
        var login = username+(country != null ? "-country-"+country : "")
            +"-session-"+session_id;
        this.Proxy.Credentials = new NetworkCredential(login, password);
    }

    protected override WebRequest GetWebRequest(Uri address)
    {
        var request = base.GetWebRequest(address) as HttpWebRequest;
        request.ConnectionGroupName = session_id;
        return request;
    }
}

class Example
{
    static void Main()
    {
        Console.WriteLine("Performing request(s)");
        var client = new Client();
        // Put full scraping sequence below:
        Console.WriteLine(client.DownloadString("http://lumtest.com/myip.json"));
        // client.DownloadString(...second request...);
    }
}



